src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/compute/v2/OpenStackComputePluginTest.java

```diff
@@ -1,858 +1,688 @@
-        Map<String, String> flavorExtraSpecs = new HashMap<String, String>();
-        flavorExtraSpecs.put(tag, value);
-        flavorExtraSpecsMap.put(OpenStackConstants.Compute.FLAVOR_EXTRA_SPECS_KEY_JSON, flavorExtraSpecs);
-        return new Gson().toJson(flavorExtraSpecsMap);
-    }
+    @Test(expected = UnexpectedException.class)
+    public void testDoRequestInstanceUnsuccessful() throws FogbowException, HttpResponseException {
+        // set up
+        ComputeOrder computeOrder = testUtils.createLocalComputeOrder();
+        Mockito.doThrow(HttpResponseException.class)
+                .when(this.clientMock).doPostRequest(Mockito.any(), Mockito.any(), Mockito.any());
 
-    private String generateInstaceId(String id) {
-        Map<String, String> instanceMap = new HashMap<String, String>();
-        instanceMap.put(OpenStackConstants.Compute.ID_KEY_JSON, id);
-        Map<String, Object> root = new HashMap<String, Object>();
-        root.put(OpenStackConstants.Compute.SERVER_KEY_JSON, instanceMap);
-        return new Gson().toJson(root);
-    }
+        List<String> networkIds = getMockedNetworkIds();
 
-    private JSONObject generateRootKeyPairJson(String keyName, String publicKey) {
-        JSONObject keypair = new JSONObject();
-        keypair.put(OpenStackConstants.Compute.NAME_KEY_JSON, keyName);
-        keypair.put(OpenStackConstants.Compute.PUBLIC_KEY_KEY_JSON, publicKey);
-        JSONObject root = new JSONObject();
-        root.put(OpenStackConstants.Compute.KEY_PAIR_KEY_JSON, keypair);
-        return root;
-    }
+        HardwareRequirements fakeRequirements = getHardwareRequirementsList().first();
 
-    private JSONObject generateJsonRequest(String imageId, String flavorId, String userData, String keyName, List<String> networkIds, String randomUUID) {
-        JSONObject server = new JSONObject();
+        // exercise
+        this.computePlugin.doRequestInstance(computeOrder, cloudUser,
+                FAKE_PROJECT_ID, flavorId, FAKE_KEY_NAME, networkIds, fakeRequirements);
 
-        // when the user doesn't provide a network, the default network should be added in the request
-        if (networkIds.isEmpty()) {
-            networkIds.add(defaultNetworkId);
-        }
+        Assert.fail();
+    }
 
-        server.put(OpenStackConstants.Compute.NAME_KEY_JSON, SystemConstants.FOGBOW_INSTANCE_NAME_PREFIX + randomUUID);
-        server.put(OpenStackConstants.Compute.IMAGE_REFERENCE_KEY_JSON, imageId);
-        server.put(OpenStackConstants.Compute.FLAVOR_REFERENCE_KEY_JSON, flavorId);
-        server.put(OpenStackConstants.Compute.USER_DATA_KEY_JSON, userData);
+    @Test
+    public void testRequestInstance() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = testUtils.createLocalComputeOrder();
 
-        JSONArray networks = new JSONArray();
+        HardwareRequirements mockRequirements = Mockito.mock(HardwareRequirements.class);
+        Mockito.doReturn(mockRequiremen
```

Comment:
Try indenting the JSON string of this and other methods of this type.

Reply:
Okay, done.